app/documents.js

```diff
@@ -172,7 +173,12 @@ export class Document extends Base {
       const doc = await this.buildDocument(model, i);
       update();
       await delay(0);
-      this.documents[model.name].push(doc);
+      // only push the document if the key is not already taken
+      /* eslint no-underscore-dangle: ["error", { "allow": ["__name", "__key"] }] */
+      if (!built_docs_map[doc.__key]) {
+        built_docs_map[doc.__key] = doc;
+        this.documents[model.name].push(doc);
+      }
     }, this.options.spinners ? 75 : 1000)
```

Comment:
After line 182 add the following code. Resting `build_docs_map` to be an empty object will free up memory to be used elsewhere.

```js
.then(() => {
  built_docs_map = {};
})
```

Reply:
Okay, I made this change. It was necessary to leave built_docs_map as a `var` to be able to re-assign it here